src/vs/workbench/api/browser/mainThreadCustomEditors.ts

```diff
@@ -138,9 +143,19 @@ export class MainThreadCustomEditors extends Disposable implements extHostProtoc
 				webviewInput.webview.options = options;
 				webviewInput.webview.extension = extension;
 
+				// If there's an old resource this was a move and we must resolve the backup at the same time as the webview
+				// This is because the backup must be ready upon model creation, and the input resolve method comes after
+				let backupID = webviewInput.backupId;
+				if (webviewInput.oldResource && !webviewInput.backupId) {
+					const backupIdentifier = { resource: webviewInput.oldResource, typeId: NO_TYPE_ID };
+					const backup = await this.workingCopyBackupService.resolve<CustomDocumentBackupData>(backupIdentifier);
+					backupID = backup?.meta?.backupId;
+					this.workingCopyBackupService.discardBackup(backupIdentifier);
```

Comment:
This operation is `async`

Reply:
Yeah, but I don't need to await it as it doesn't matter when the backup gets discarded. At least that was my thought.